### PR TITLE
consider some errors non-retryable closes #124

### LINKED ISSUE #124: Gracefully handle failing split search
Right now if a single split fails, we fail. We need to at least retry this split, or return partial results and point out the failed split.

### PR DESCRIPTION
**Pull request by PSeitz**
Consider missing index and invalid query as non retryable

### CODE REVIEW

**Review by fulmicoton — CHANGES_REQUESTED** [review]
Can we have a unit test?

File: quickwit-search/src/leaf.rs [hunk 1]
```diff
@@ -178,10 +178,19 @@ pub async fn leaf_search(
...
-        .extend(errors.iter().map(|(split_id, err)| SplitSearchError {
-            split_id: split_id.to_string(),
-            error: format!("{:?}", err),
-            retryable_error: true,
+        .extend(errors.iter().map(|(split_id, err)| {
+            let retryable_error = match err {
+                SearchError::IndexDoesNotExist { index_id: _ } | SearchError::InvalidQuery(_) => {
+                    false
+                }
+                _ => true,
```
Comment (fulmicoton) [inline quickwit-search/src/leaf.rs:186]: Can you put the list of items instead of using the default pattern?

**Review by fulmicoton — CHANGES_REQUESTED** [review]
See comment on the matched enum

File: quickwit-search/src/leaf.rs (hunk 1 above)
Comment (PSeitz) [reply quickwit-search/src/leaf.rs]: I moved it to a a method to the enum,  so it should be easier to spot

**Comment by fulmicoton** [comment]
@fmassot @PSeitz Is this PR now obsolete?

**Comment by PSeitz** [comment]
I think it's still valid

**Comment by fmassot** [comment]
I would not keep the code at the search error level and define retry able errors at the retry policy level.

**Comment by fulmicoton** [comment]
@PSeitz Closing the PR.

If this is valuable, discuss with Francois where to put that and **add unit tests**.

**Pull request closed without merging**